snappy-core/src/main/scala/org/apache/spark/sql/execution/datasources/StoreDataSourceStrategy.scala

```diff
@@ -77,9 +77,11 @@ private[sql] object StoreDataSourceStrategy extends Strategy with Logging {
       filterPredicates,
       numPartition,
       partitionColumns,
-      (requestedColumns, pushedFilters) => {
-        scanBuilder(requestedColumns, selectFilters(pushedFilters).toArray)
+      (requestedColumns, _, pushedFilters) => {
+        scanBuilder(requestedColumns, pushedFilters.toArray)
       })
+
```

Comment:
This the area of code , which signals which predicates have been pushed down , so that filter does not apply on them again. I have that in mind and going forward will resolve this.